src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,83 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const handleClose = () => {
+    dispatch(setCurrentTodo(null));
+  };
+
+  const fetchUserData = async (userId: number) => {
+    setIsLoading(true);
+    try {
+      const userData = await getUser(userId);
+
+      setUser(userData);
+    } finally {
+      setIsLoading(false);
+    }
```

Comment:
The `fetchUserData` function does not handle errors that might occur during the API call. Consider adding a `catch` block to handle potential errors and provide feedback to the user, such as setting an error state or displaying an error message.